metric/system/process/process_common.go

```diff
@@ -33,9 +34,42 @@ import (
 	sysinfo "github.com/elastic/go-sysinfo"
 )
 
-// ProcsMap is a map where the keys are the names of processes and the value is the Process with that name
+//ProcsMap is a convinence wrapper for the oft-used ideom of map[int]ProcState
 type ProcsMap map[int]ProcState
 
+// ProcsTrack is a thread-safe wrapper for a process Stat object's internal map of processes.
+type ProcsTrack struct {
+	pids ProcsMap
+	mut  sync.Mutex
+}
+
+func NewProcsMap() *ProcsTrack {
```

Comment:
```suggestion
func NewProcsTrack() *ProcsTrack {
```

nit: The constructor name should match the type name.

Reply:
Similarly I see the receiver is named `pm` in the methods below, did you switch the name from `ProcsMap` to `ProcsTrack` and only partially update the naming?